src/include/ipc4/copier.h

```diff
@@ -265,6 +265,7 @@ struct copier_data {
 	struct host_data *hd;
 	bool ipc_gtw;
 	struct dai_data *dd[IPC4_ALH_MAX_NUMBER_OF_GTW];
+	struct ipcgtw_data *ipcgtw_data[IPC4_ALH_MAX_NUMBER_OF_GTW];
```

Comment:
because if without array, then the free like function:

for (i = 0; i < cd->endpoint_num; i++) {
				cd->endpoint[i]->drv->ops.free(cd->endpoint[i]);					 
                                 ipcgtw_zephyr_free(cd->ipcgtw_data[i]);
}

seems gtw test, cd->endpoint_num may >1, previously, I did not add array, and QB will report one error, after add it, QB passed.

Reply:
@btian1 please remove the loop for ipc_gtw. There's really no need

Reply:
ok, let's wait QB free, I will test again.